mido/midifiles/units.py

```diff
@@ -17,7 +17,7 @@ def second2tick(second, ticks_per_beat, tempo):
     note) and tempo (microseconds per beat).
     """
     scale = tempo * 1e-6 / ticks_per_beat
-    return second / scale
+    return int(round(second / scale))
```

Comment:
This change will break a program of mine where I let the user decrease the PPQ of MIDI files.

```py
ticks = int(round(mido.second2tick(message.time, ppq, tempo)))
assert ticks > 0 or message.time == 0, "New PPQ is too low."
```

Reply:
It would be good to see which part of the assert fails (I am not sure if this makes any difference, though), then it's easier to discuss this further.

The rationale behind this change is that "the MIDI standard" requires QQN to be an integer. And until now this function could return a float value.

I am not sure why ticks should be greater 0 or message timings should not be 0. Both are legitimate values as far as I can see. What's the intention behind these checks?

The problem seems to be that different Python versions handle the /-operator differently. Anyways, I see that this change breaks code (although I never thought it would), but IMHO the question remains if your assertion assumes the "right thing".

Reply:
> The rationale behind this change is that "the MIDI standard" requires QQN to be an integer. And until now this function could return a float value.

How should we handle the case when second will be truncated to zero for the given PPQ and tempo (or heavily shifted in time, `ceil` doesn't really help)?

> What's the intention behind these checks?

The assertion checks that there will be at least one tick with the given `ppq` when `message.time` is larger than 0.0 seconds. 

When PPQ is chosen too low there could be messages that don't move the tick count at all even though they should, which causes a slight global drift AFAIK.

The problem stems from MAPS having ridiculously high PPQ (https://github.com/craffel/pretty-midi/issues/112), while just setting it to 960 is fine because no message actually seems to require that much time resolution to fit into a piano roll (although the messages might shift just slightly forward and/or backwards in time, which I don't bother to check).

Reply:
Btw, what does QQN stand for?

Reply:
Typo, TQN = ticks per quarter note.

Reply:
High PPQ (e.g. in the MAPS dataset) should be fine, since they result in large tick values. On the other hand, having low PPQ values _is_ a (known) problem, but I am still not sure how this is affected by the proposed change.

> The assertion checks that there will be at least one tick with the given ppq when message.time is larger than 0.0 seconds.

Ok, but I don't see how moving the `int(round(...))` thing inside the function should alter the result since you apply exactly the same rounding and casting to int by yourself. Are you sure that this change is the culprit and not something else?

Reply:
> Ok, but I don't see how moving the int(round(...)) thing inside the function should alter the result since you apply exactly the same rounding and casting to int by yourself.

Haha, you're absolutely right... Sorry. :laughing: 

![image](https://user-images.githubusercontent.com/1595907/31069390-904edea0-a75c-11e7-8150-016a2087ef3d.png)

Anyway, could we have something like this internally maybe?

```python
if second != 0 and tick == 0:
    warnings.warn(f"{second} seconds is zero ticks at {ppq} PPQ.")
```